sstables/sstables.hh

```diff
@@ -142,6 +142,10 @@ inline std::string_view state_to_dir(sstable_state state) {
         return quarantine_dir;
     case sstable_state::upload:
         return upload_dir;
+    case sstable_state::unlinked:
+        // this should never be called and is here for completeness
+        SCYLLA_ASSERT(false);
+        return "";
     }
```

Comment:
I'd rather not introduce a unlinked to sstable state, since as you can see, it's a bit more specific than a generic state. And I am sure we can break this assumption (e.g. compaction racing with streaming) :-) 

I'd prefer if we could confine the unlinked tracking in manager, by introducing a 'set_type _unlinked', which we update in on_unlink() and deactivate().